### ISSUE TITLE
General Question: General setup with batch of text input

### ISSUE THREAD

**Issue by nikky4D** [openai/CLIP#17]
Could you recommend a way to setup batch processing with a bunch of text tokens per class?

For example, I am setting up a task where I have 3 classes, and 2 text sentences per class. For instance, for classes=[daisy, daffodil, lavender], I have the following sentences: 

- for class daisy: "pushing up daisy", "daisy"
- for class daffodil: "planting daffodils", "daffodil"
- for class lavender: "smells like lavender", "lavender is good for skin"

The task is then given an image, which class best describes the image. 
Right now, I am tokenizing each sentence in the classes. This gives me a batch of 3 x 2 x 77:  3 classes by 2 sentences by 77 context number. 
Then for an image, I would iterate over the 2 sentences, doing logits_per_img, logits_per_text = model(image, batch of text[:, i, :]) where i is index of a sentence in the list of sentence descriptors for the class list. 

After this, I take the softmax, then the maximum. 

My question is: is there an efficient batch processing method I can use for this? And can I increase the context number for longer sentences?

**Comment by thoppe** [comment]
Not the authors, but the context window is baked into the pre-trained model, it's not adjustable unless you train a new model.

FWIW you can compute the sentence vector _once_ and then reuse them for a large batch of images. See https://github.com/thoppe/alph-the-sacred-river for some examples of where I do just that.

**Comment by nikky4D** [comment]
Thanks. I'll try it out.
